client/src/components/ResourceManager.jsx

```diff
@@ -54,34 +63,58 @@ const SidebarCategory = (props: Props) => {
         <Menu.Item key={subCategory} className="subcategory">
           {subCategory}
           <div>
-            <EditOutlined />
-            <CloseOutlined style={{ color: '#FF0000' }} />
+            <EditCategoryModal
+              modalType="rename"
+              categoryType="subcategory"
+              subcategoryName={subCategory}
+              id={categoryIds[categoryName]}
+              categoryName={categoryName}
+              fetchCategories={fetchCategories}
+            />
+            <EditCategoryModal
+              modalType="delete"
+              categoryType="subcategory"
+              subcategoryName={subCategory}
+              id={categoryIds[categoryName]}
+              categoryName={categoryName}
+              fetchCategories={fetchCategories}
+            />
           </div>
         </Menu.Item>
       ))}
       <Menu.Item key="Add Subcategory" className="subcategory">
```

Comment:
Can the ID for this be something like `categoryIds[categoryName]-add-subcategory`? Generally you want to keep keys unique so React can do some cool optimization stuff :) This won't be unique since it's inside a `map()`